plugins/aws/fix_plugin_aws/access_edges.py

```diff
@@ -63,13 +63,27 @@ def find_allowed_action(policy_document: PolicyDocument, service_prefix: str) ->
     return allowed_actions
 
 
+def find_non_service_actions(resource_arn: str) -> Set[IamAction]:
+    splitted = resource_arn.split(":")
+    service_prefix = splitted[2]
+    if service_prefix == "iam":
+        resource_type = splitted[5]
```

Comment:
Let's avoid accessing arrays by index without checking the length or wrapping it in a try/catch.